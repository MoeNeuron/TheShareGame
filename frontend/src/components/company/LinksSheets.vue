<template>
  <div class="card card-body bg-light">
    <table class="table borderless" id="data-table">
      <tbody>
      <tr style="border-top: hidden;">
        <td>
          <router-link :to="{name: 'order', params: {isin: isin}}">
            Place Order
          </router-link>
        </td>
        <td>
          <router-link :to="{name: 'orderbook', params: {isin: isin,
           companyName: companyName}}">
            Orderbook
          </router-link>
        </td>
      </tr>
      <tr style="border-top: hidden;">

        <td>
          <router-link :to="{name: 'statementOfAccount', params: {isin: isin,
           companyName: companyName}}">
            Statement of Account
          </router-link>
        </td>
        <td>
          <router-link :to="{name: 'tradesCompany', params: {isin: isin,
          companyName: companyName}}">
            Trades
          </router-link>
        </td>
      </tr>
      <tr style="border-top: hidden;">
        <td v-if="user !== null">
          <router-link :to="{name: 'messages', query: {user_id: user.id}}">
          Message {{user.username}}
          </router-link>
        </td>
        <td>
          <router-link :to="{name: 'articlesCompany', params: {isin: isin,
           companyName: companyName}}">
          Articles
            </router-link>
        </td>
      </tr>
      <tr style="border-top: hidden;">
        <td>
          <router-link :to="{name: 'depot', params: {isin: isin,
           companyName: companyName}}">
            Depot - {{this.depot.length}}
          </router-link>
        </td>
        <td>
          <router-link :to="{name: 'shareholders', params: {isin: isin,
           companyName: companyName}}">
          Shareholders - {{this.shareholders.length}}
          </router-link>
        </td>
      </tr>
      <tr style="border-top: hidden;">
        <td>
          <router-link :to="{name: 'buyerSeller', params: {isin: isin,
          companyName: companyName}}">
            Buyer/Seller
          </router-link>
        </td>
        <td v-if="user !== null">
          CEO:
          <router-link :to="{name: 'profile', params: {id: user.id}}">
            {{user.username}}
          </router-link>
        </td>
      </tr>
      </tbody>
    </table>
  </div>
</template>

<script>
export default {
  name: 'LinksSheets',
  props: {
    isin: String,
    user: Object,
    companyName: String,
    shareholders: {
      type: Array,
      default() {
        return [];
      },
    },
    depot: {
      type: Array,
      default() {
        return [];
      },
    },

  },
};
</script>

<style scoped>

</style>
