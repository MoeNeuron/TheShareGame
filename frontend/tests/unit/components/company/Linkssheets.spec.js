import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import LinksSheets from '@/components/company/LinksSheets.vue';
import mockRouter from '../../mockRouter';


const localVue = createLocalVue();
localVue.use(VueRouter);
const router = mockRouter.mock();

describe('LinksSheet', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(LinksSheets, {
      localVue,
      router,
      propsData: {
        isin: 'JP000001',
        user: { id: 1, username: 'Son Goku' },
      },
    });
  });

  it('matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
