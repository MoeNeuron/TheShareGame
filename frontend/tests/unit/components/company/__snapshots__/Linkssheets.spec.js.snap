// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksSheet matches Snapshot 1`] = `
<div class="card card-body bg-light">
  <table id="data-table" class="table borderless">
    <tbody>
      <tr style="border-top: hidden;">
        <td><a href="#/company/JP000001/order" class="">
            Place Order
          </a></td>
        <td><a href="#/company/JP000001/orderbook" class="">
            Orderbook
          </a></td>
      </tr>
      <tr style="border-top: hidden;">
        <td><a href="#/company/JP000001/statement_of_account" class="">
            Statement of Account
          </a></td>
        <td><a href="#/company/JP000001/trades" class="">
            Trades
          </a></td>
      </tr>
      <tr style="border-top: hidden;">
        <td><a href="#/messages?user_id=1" class="">
            Message Son Goku
          </a></td>
        <td><a href="#/company/JP000001/articles" class="">
            Articles
          </a></td>
      </tr>
      <tr style="border-top: hidden;">
        <td><a href="#/company/JP000001/depot" class="">
            Depot - 0
          </a></td>
        <td><a href="#/company/JP000001/shareholders" class="">
            Shareholders - 0
          </a></td>
      </tr>
      <tr style="border-top: hidden;">
        <td><a href="#/company/JP000001/buyerSeller" class="">
            Buyer/Seller
          </a></td>
        <td>
          CEO:
          <a href="#/profile/1" class="">
            Son Goku
          </a></td>
      </tr>
    </tbody>
  </table>
</div>
`;
